src/components/display/Table.jsx

```diff
@@ -1,63 +1,50 @@
-import React, {Component} from 'react';
+import React, {useState, useEffect} from 'react';
 import PropTypes from 'prop-types';
-
 import {Box, Table as MuiTable, TableContainer, TablePagination} from '@material-ui/core';
-
 import {TableBody, TableHead} from '../../fragments/TableContent';
 
 /**
  * Table component
  */
-export default class Table extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      page: 0,
-      rows: props.rows,
-      rowsPerPage: props.rowsPerPageOptions[0]
-    };
-  }
+const Table = (props) => {
+  const {id, columns, rowsPerPageOptions, rows} = props;
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(rowsPerPageOptions[0]);
 
-  handlePageChange = (event, value) => {
-    this.setState({page: value});
-  };
+  // Make sure state remain in sync with received props
+  useEffect(() => {
+    props.setProps({rows: rows});
```

Comment:
❓ 

A `Table` is just a display component for a data grid, not requiring a user input to trigger Dash callbacks.

Why is it required to update the `rows` props in the parent Dash renderer via the use of the `setProps` props (rather than just maintaining the data as a component state)?

Reply:
Good point, wrote it this way to be uniform with the others, will change to **setRows**.